File: head/cddl/contrib/dtracetoolkit/Code/Ruby/func_abc.rb
#!./ruby -w

def func_c
  print "Function C\n"
  sleep 1
end

def func_b
  print "Function B\n"
  sleep 1
  func_c
end

def func_a
  print "Function A\n"
  sleep 1
  func_b
end

func_a

File: head/cddl/contrib/dtracetoolkit/Code/Perl/func_abc.pl
#!./perl -w

sub func_c {
    print "Function C\n";
    sleep 1;
}

sub func_b {
    print "Function B\n";
    sleep 1;
    func_c();
}

sub func_a {
    print "Function A\n";
    sleep 1;
    func_b();
}

func_a();
